code/modules/mob/living/carbon/xenomorph/abilities/general_abilities.dm

```diff
@@ -284,7 +284,7 @@
 
 /datum/action/xeno_action/onclick/toggle_long_range/can_use_action()
 	var/mob/living/carbon/xenomorph/xeno = owner
-	if(xeno && !xeno.is_mob_incapacitated() && !xeno.buckled)
+	if(xeno?.check_state())
```

Comment:
This results in resting always spamming you with `We cannot do this in our current state.`

As long as the ability properly ends when resting, just the check_state when trying to use_ability is likely sufficient.